build/vsts-ci.yml

```diff
@@ -2,309 +2,317 @@
-      inputs:
-        pathToPublish: $(Build.SourcesDirectory)/artifacts/pkgassets
-        artifactName: pkgassets
-        artifactType: container
-
-    - template: /eng/common/templates/steps/generate-sbom.yml
-
-    # Terminate all dotnet build processes.
-    - script: $(Build.SourcesDirectory)/.dotnet/dotnet.exe build-server shutdown
-      displayName: Dotnet Server Shutdown
-
-  ################################################################################
-  - job: Windows_x86
-  ################################################################################
-    variables:
-      _TeamName: DotNetCore
-    pool:
-      name: NetCore1ESPool-Internal
-      demands: ImageOverride -equals windows.vs2019.amd64
-    steps:
-    # Only build native assets to avoid conflicts.
-    - script: ./build.cmd -projects $(Build.SourcesDirectory)/src/Native/Native.proj -configuration $(BuildConfig) /p:TargetArchitecture=x86 /p:CopyPackageAssets=true
-      displayName: Build Native Assets
-
-    - task: PublishBuildArtifacts@1
-      displayName: Publish Windows_x86 package assets
-      inputs:
-        pathToPublish: $(Build.SourcesDirectory)/artifacts/pkgassets
-        artifactName: pkgassets
-        artifactType: container
-
-    - template: /eng/common/templates/steps/generate-sbom.yml
-
-    # Terminate all dotnet build processes.
-    - script: $(Build.SourcesDirectory)/.dotnet/dotnet.exe build-server shutdown
-      displayName: Dotnet Server Shutdown
-
-  ################################################################################
-  - job: Windows_x64
-  ################################################################################
-    variables:
-      Codeql.SkipTaskAutoInjection: False  # run CodeQL in this job
-      _TeamName: DotNetCore
+resources:
+  repositories:
+  - repository: 1ESPipelineTemplates
+    type: git
+    name: 1ESPipelineTemplates/1ESPipelineTemplates
+    ref: refs/tags/release
+
+extends:
+  template: v1/1ES.Official.PipelineTemplate.yml@1ESPipelineTemplates
+  parameters:
+    featureFlags:
+      autoBaseline: true
+    sdl:
+      binskim:
+        enabled: true
+        scanOutputDirectoryOnly: true
     pool:
-      name: NetCore1ESPool-Internal
-      demands: ImageOverride -equals windows.vs2019.amd64
-    steps:
-
-    # Build both native and managed assets.
-    - script: ./build.cmd -configuration $(BuildConfig) /p:TargetArchitecture=x64 /p:CopyPackageAssets=true
+      name: $(DncEngInternalBuildPool)
+      image: 1es-windows-2019-pt
+      os: windows
+    containers:
+      CentosContainer: 
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8-mlnet
+      UbuntuCrossArmContainer: 
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-mlnet-cross-arm
+      UbuntuCrossArm64Container:
+        image: mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-mlnet-cross-arm64
+    customBuildTags:
+    - ES365AIMigrationTooling
```

Comment:
Do we need to keep this tag?

Reply:
Good question. I didn't actually use AI migration tooling in this case (I took what I learned from conversion tooling on `node` & applied it more minimally). It doesn't seem to be in the actual docs so probably isn't really needed.